packages/zod/src/schemas.ts

```diff
@@ -0,0 +1,2402 @@
+  inst.xid = (params) => inst.check(core._xid(ZodXID, params));
+  inst.ksuid = (params) => inst.check(core._ksuid(ZodKSUID, params));
+  // inst.ip = (params) => {
+  //   const version = (params ?? {} as any)?.version;
+  //   if (version === "v4") return inst.check(core._ipv4(ZodIPv4, params));
+  //   if (version === "v6") return inst.check(core._ipv6(ZodIPv6, params as any));
+  //   return union([inst.ipv4(params), inst.ipv6(params as any)]);
+  // };
+  inst.ipv4 = (params) => inst.check(core._ipv4(ZodIPv4, params));
+  inst.ipv6 = (params) => inst.check(core._ipv6(ZodIPv6, params));
+  inst.cidrv4 = (params) => inst.check(core._cidrv4(ZodCIDRv4, params));
+  inst.cidrv6 = (params) => inst.check(core._cidrv6(ZodCIDRv6, params));
+  inst.e164 = (params) => inst.check(core._e164(ZodE164, params));
+
+  // iso
+  inst.datetime = (params) => inst.check(iso.datetime(params as any));
+  inst.date = (params) => inst.check(iso.date(params as any));
+  inst.time = (params) => inst.check(iso.time(params as any));
+  inst.duration = (params) => inst.check(iso.duration(params as any));
+
+  // validations
+  inst.regex = (...args) => inst.check(checks.regex(...args));
+  inst.includes = (...args) => inst.check(checks.includes(...args));
+  inst.startsWith = (params) => inst.check(checks.startsWith(params));
+  inst.endsWith = (params) => inst.check(checks.endsWith(params));
+  inst.min = (...args) => inst.check(checks.minLength(...args));
+  inst.max = (...args) => inst.check(checks.maxLength(...args));
+  inst.length = (...args) => inst.check(checks.length(...args));
+  inst.nonempty = (...args) => inst.check(checks.minLength(1, ...args));
+  inst.lowercase = (params) => inst.check(checks.lowercase(params));
+  inst.uppercase = (params) => inst.check(checks.uppercase(params));
+
+  // transforms
+  inst.trim = () => inst.check(checks.trim());
+  inst.normalize = (...args) => inst.check(checks.normalize(...args));
+  inst.toLowerCase = () => inst.check(checks.toLowerCase());
+  inst.toUpperCase = () => inst.check(checks.toUpperCase());
+});
+
+export function string(params?: string | core.$ZodStringParams): ZodString {
+  return core._string(ZodString, params) as any;
+}
+
+// ZodStringFormat
+export interface ZodStringFormat<Format extends core.$ZodStringFormats = core.$ZodStringFormats> extends ZodString {
+  _zod: core.$ZodStringFormatInternals<Format>;
+}
+export const ZodStringFormat: core.$constructor<ZodStringFormat> = /*@__PURE__*/ core.$constructor(
+  "ZodStringFormat",
+  (inst, def) => {
+    core.$ZodStringFormat.init(inst, def);
+    ZodString.init(inst, def);
+  }
+);
+
+// ZodEmail
+export interface ZodEmail extends ZodStringFormat<"email"> {
+  _zod: core.$ZodEmailInternals;
+}
+export const ZodEmail: core.$constructor<ZodEmail> = /*@__PURE__*/ core.$constructor("ZodEmail", (inst, def) => {
+  // ZodStringFormat.init(inst, def);
+  core.$ZodEmail.init(inst, def);
+  ZodType.init(inst, def);
+});
```

Comment:
All shorthand classes (formats) are missing methods for additional refinements.
See #4162 

![image](https://github.com/user-attachments/assets/595acd31-5ded-4ade-b921-561240f1dbb9)